src/Server.cpp

```diff
@@ -38,19 +37,29 @@ inline static void kana_output_xml(const char *kana, struct evbuffer *buffer) {
 } 
 
 inline static void parse_output_xml_header(struct evbuffer *buffer) {
-    evbuffer_add_printf(buffer, "<parse>");
+    evbuffer_add_printf(buffer, "<parse>\n");
 }
 
 inline static void parse_output_xml_footer(struct evbuffer *buffer) {
     evbuffer_add_printf(buffer, "</parse>\n");
 }
 
-inline static void furigana_output_xml_header(struct evbuffer *buffer) {
-    evbuffer_add_printf(buffer, "<furigana>");
+inline static void cdata_output_xml(const char *string, struct evbuffer *buffer) {
+    evbuffer_add_printf(buffer, "<![CDATA[%s]]>", string);
 }
 
-inline static void furigana_output_xml_footer(struct evbuffer *buffer) {
-    evbuffer_add_printf(buffer, "</furigana>\n");
+inline static void reading_output_xml(std::pair<std::string, std::string> text,
+                                      struct evbuffer *buffer) {
+    evbuffer_add_printf(buffer, "<reading furigana=\"%s\"><![CDATA[%s]]></reading>",
+                        text.second.c_str(), text.first.c_str());
```

Comment:
here for example ,   
   
   * the  `buffer` variable should be on a separate line 
  *  each parameter should be on a line
  * `text.second.c_str() ` is aligned  with the method name (it should just be indented by 1)